specifyweb/frontend/js_src/lib/components/QueryBuilder/FieldFilter.tsx

```diff
@@ -440,16 +449,21 @@ export const queryFieldFilters: RR<
     label: queryText.equal(),
     description: undefined,
     renderPickList: true,
+    types: ['text', 'number', 'date', 'id'],
     component: SingleField,
     hasParser: true,
-    types: ['text', 'number', 'date', 'id'],
   },
   greater: {
     id: 2,
     label: queryText.greaterThan(),
     description: undefined,
     renderPickList: false,
-    types: ['number', 'date', 'id'],
+    types: filterArray([
+      showComparisonOperatorsForString() ? 'text' : undefined,
```

Comment:
`showComparisonOperatorsForString` using an f.store is nice, but it's not enough to fix the bug
the `queryFieldFilters` is still created only once, and in the global scope - so it will call `showComparisonOperatorsForString` before preferences are fetched

you need to wrap `queryFieldFilters` in f.store too